platforms/documentation/docs/src/docs/userguide/dep-man/declaring-repositories/supported_repository_types.adoc

```diff
@@ -0,0 +1,160 @@
+// Copyright (C) 2023 Gradle, Inc.
+//
+// Licensed under the Creative Commons Attribution-Noncommercial-ShareAlike 4.0 International License.;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://creativecommons.org/licenses/by-nc-sa/4.0/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[repository-types]]
+= Repository Types
+
+Gradle supports various sources for resolving dependencies, accommodating different formats and connectivity methods.
+You can resolve dependencies from:
+
+- <<#sub:flat_dir_resolver,Local (flat) directories>>
+- <<#sec:maven_repo,Maven-compatible artifact repositories>> (e.g., Maven Central)
+- <<#sec:ivy_repositories,Ivy-compatible artifact repositories>> (including custom layouts)
+
+[[sub:flat_dir_resolver]]
+== Flat directory repository
+
+Some projects store dependencies on a shared drive or within the project’s source code rather than using a binary repository.
+To use a flat filesystem directory as a repository, you can configure it like this:
+
+====
+include::sample[dir="snippets/artifacts/defineRepository/kotlin",files="build.gradle.kts[tags=flat-dir-multi]"]
+include::sample[dir="snippets/artifacts/defineRepository/groovy",files="build.gradle[tags=flat-dir-multi]"]
+====
+
+This configuration adds repositories that search specified directories for dependencies.
+
+Flat directory repositories do not support metadata formats like Ivy XML or Maven POM files.
+Instead, Gradle dynamically generates a module descriptor based on the presence of artifacts, without any dependency information.
+
+Gradle prefers modules with real metadata over those generated by flat directory repositories, so flat directories cannot override artifacts with real metadata from other declared repositories.
+For instance, if Gradle finds `jmxri-1.2.1.jar` in a flat directory and `jmxri-1.2.1.pom` in another repository, it will use the metadata from the latter.
```

Comment:
```suggestion
When resolving dependencies from a flat dir repo, Gradle dynamically generates adhoc dependency metadata based on the presence of artifacts.
Gradle prefers modules with real metadata over those generated by flat directory repositories. For this reason, flat directories cannot override artifacts with real metadata from other declared repositories.

For instance, if Gradle finds `jmxri-1.2.1.jar` in a flat directory and `jmxri-1.2.1.pom` in another repository, it will use the metadata from the latter.
```

Reply:
Done.